lib/node_modules/@stdlib/lapack/base/dlange/lib/base.js

```diff
@@ -0,0 +1,362 @@
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @param {Float64Array} work - work array, should have `N` indexed elements if row-major layout is used
+* @param {integer} strideWork - stride length of `work`
+* @param {NonNegativeInteger} offsetWork - starting index of `work`
+* @returns {number} required norm value
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+* var work = new Float64Array( 4 );
+*
+* var out = oneNorm( 3, 4, A, 4, 1, 0, work, 1, 0 );
+* // returns 33.0
+*/
+function oneNorm( M, N, A, strideA1, strideA2, offsetA, work, strideWork, offsetWork ) { // eslint-disable-line max-len
+	var value;
+	var temp;
+	var ia1;
+	var ia2;
+	var sum;
+	var iw;
+	var i;
+	var j;
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		iw = offsetWork;
+		for ( i = 0; i < N; i++ ) {
+			work[ iw ] = 0.0;
+			iw += strideWork;
+		}
+
+		ia1 = offsetA;
+		for ( j = 0; j < M; j++ ) {
+			ia2 = 0;
+			iw = offsetWork;
+			for ( i = 0; i < N; i++ ) {
+				work[ iw ] += abs( A[ ia1 + ia2 ] );
+				iw += strideWork;
+				ia2 += strideA2;
+			}
+			ia1 += strideA1;
+		}
+
+		value = 0.0;
+
+		iw = offsetWork;
+		for ( i = 0; i < N; i++ ) {
+			temp = work[ iw ];
+			if ( value < temp || isnan( temp ) ) {
+				value = temp;
+			}
+			iw += strideWork;
+		}
+	} else {
+		value = 0.0;
+		ia1 = offsetA;
+		for ( j = 0; j < N; j++ ) {
+			sum = dasum( M, A, strideA1, ia1 );
+			if ( value < sum || isnan( sum ) ) {
+				value = sum;
+			}
+			ia1 += strideA2;
+		}
+	}
+
+	return value;
+}
+
+/**
+* Returns the absolute value of the maximum element of a real matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @returns {number} required norm value
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+*
+* var out = maxAbs( 3, 4, A, 4, 1, 0 );
+* // returns 12.0
+*/
+function maxAbs( M, N, A, strideA1, strideA2, offsetA ) {
+	var value;
+	var temp;
+	var da0;
+	var da1;
+	var ia;
+	var sa;
+	var sh;
+	var S0;
+	var S1;
+	var o;
+	var i;
+	var j;
+
+	value = 0.0;
+
+	// Resolve the loop interchange order:
+	o = loopOrder( [ M, N ], [ strideA1, strideA2 ] );
```

Comment:
We typically use `loopOrder` when implementing a blocked algorithm. Instead, I suggest doing something similar to  L281 below and use an explicit `if` statement.